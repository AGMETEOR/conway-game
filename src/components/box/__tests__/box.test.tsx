import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Box from '../box';

afterEach(cleanup);


describe('Box', () => {
  const props = {
    row: 5,
    col: 5,
    boxClass: "",
    boxId: 'test-id',
    selectBox: jest.fn(),
  }
  
  it('should match stored snapshot', () => {
    const { asFragment } = render(<Box {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  it('should call selectBox when clicked', () => {
    const { getByTestId } = render(<Box {...props} />);
  
    // component test id is prepared using the boxid
    fireEvent.click(getByTestId(props.boxId));
    expect(props.selectBox).toHaveBeenCalled();
  });

});

