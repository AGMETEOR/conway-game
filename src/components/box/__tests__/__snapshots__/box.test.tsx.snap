// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should match stored snapshot 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="test-id"
    id="test-id"
  />
</DocumentFragment>
`;
